tests/Unit/Time/Test_TakeStep.cpp

```diff
@@ -0,0 +1,124 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/Time/TimeSteppers/TimeStepperTestUtils.hpp"
+#include "Time/StepChoosers/StepToTimes.hpp"
+#include "Time/StepControllers/BinaryFraction.hpp"
+#include "Time/Tags.hpp"
+#include "Time/TakeStep.hpp"
+#include "Time/Time.hpp"
+#include "Time/TimeSequence.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/Literals.hpp"
+
+namespace {
+struct EvolvedVariable : db::SimpleTag {
+  using type = DataVector;
+};
+
+struct Metavariables {
+  struct system {
+    using variables_tag = EvolvedVariable;
+  };
+  using component_list = tmpl::list<>;
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Time.TakeStep", "[Unit][Time]") {
+  using step_chooser_list = tmpl::list<StepChoosers::Registrars::StepToTimes>;
+  const Slab slab{0.0, 0.01};
+  const TimeDelta time_step = slab.duration() / 2;
+  std::unique_ptr<TimeSequence<double>> times{
+      std::make_unique<TimeSequences::Specified<double>>(
+          std::vector<double>{0.005, 0.0075, 0.00875, 0.01})};
```

Comment:
I don't understand why this test passes.  I think the floating point error in these values should cause the step to drop compared to the analytic value.  (Because `0.0075 - 0.005 < 0.01 / 4`)